codebuild/bin/install_al2_dependencies.sh

```diff
@@ -16,6 +16,11 @@
 set -eu
 source ./codebuild/bin/s2n_setup_env.sh
 
+if [[ ${DISTRO} != "amazon linux" ]]; then
+    echo "Target AL2, but running on $DISTRO: Nothing to do."
```

Comment:
The idea was to make the buildspec.yml so generic that it can run everywhere; moving the checks for which platform we're on down the stack.  Removing the buildspec changes from the PR makes this less obvious :-(.